src/nvim/ops.c

```diff
@@ -1408,6 +1408,10 @@ int op_delete(oparg_T *oap)
     }
 
     if (oap->regname == 0) {
+      if (reg == NULL) {
+        // Guard against NULL reg
+        return FAIL;
+      }
```

Comment:
if `reg` is null here, it's a bug. So I agree it should `abort()` or `assert(false)`.